config/application.rb

```diff
@@ -144,13 +144,20 @@ class Application < Rails::Application
     config.view_component.show_previews = IdentityConfig.store.component_previews_enabled
     if IdentityConfig.store.component_previews_enabled
       require 'lookbook'
+      require 'component_preview_csp'
 
       config.view_component.preview_controller = 'ComponentPreviewController'
       config.view_component.preview_paths = [Rails.root.join('spec', 'components', 'previews')]
       config.view_component.default_preview_layout = 'component_preview'
       config.lookbook.auto_refresh = false
       config.lookbook.project_name = "#{APP_NAME} Component Previews"
       config.lookbook.ui_theme = 'blue'
+      if IdentityConfig.store.component_previews_embed_frame_ancestors.present?
+        # so we can embed a lookbook component into the dev docs
+        config.lookbook.preview_embeds.policy = "ALLOWALL"
+        # lookbook strips out CSP, this brings it back so we aren't so permissive
+        config.middleware.insert_after ActionDispatch::Static, ComponentPreviewCsp
+      end
```

Comment:
I think we could avoid loading the `ComponentPreviewCsp` file unless we know we're going to use it.

```suggestion

      config.view_component.preview_controller = 'ComponentPreviewController'
      config.view_component.preview_paths = [Rails.root.join('spec', 'components', 'previews')]
      config.view_component.default_preview_layout = 'component_preview'
      config.lookbook.auto_refresh = false
      config.lookbook.project_name = "#{APP_NAME} Component Previews"
      config.lookbook.ui_theme = 'blue'
      if IdentityConfig.store.component_previews_embed_frame_ancestors.present?
        # so we can embed a lookbook component into the dev docs
        config.lookbook.preview_embeds.policy = "ALLOWALL"
        # lookbook strips out CSP, this brings it back so we aren't so permissive
        require 'component_preview_csp'
        config.middleware.insert_after ActionDispatch::Static, ComponentPreviewCsp
      end
```